### DISCUSSION TITLE
Some PackageRule are not applied as expected

### DISCUSSION THREAD

**Discussion by tom-agotchi-318** [renovatebot/renovate#9568] [Request Help]
Hi Renovate team!

I just created a shared configuration package for our repositories. There are 2 rules, that are not applied and I have no idea why.

Here is our shared config (in package.json):

```
  "renovate-config": {
    "default": {
      "assignAutomerge": true,
      "recreateClosed": true,
      "stabilityDays": 5,
      "reviewers": [
        "censored2",
        "censored2"
      ],
      "extends": [
        ":automergeRequireAllStatusChecks",
        ":combinePatchMinorReleases",
        ":enableVulnerabilityAlertsWithLabel(vulnerability)",
        ":ignoreModulesAndTests",
        ":ignoreUnstable",
        ":label(depedendencies)",
        ":pinVersions",
        ":prConcurrentLimit10",
        ":prNotPending",
        ":semanticCommits",
        ":separateMajorReleases",
        ":updateNotScheduled",
        "group:monorepos",
        "group:symfony",
        "helpers:disableTypesNodeMajor",
        "schedule:daily"
      ],
      "packageRules": [
        {
          "packageNames": [
            "node"
          ],
          "allowedVersions": "14.16.0"
        },
        {
          "packageNames": [
            "semantic-release"
          ],
          "allowedVersions": "17.0.0"
        },
        {
          "packageNames": [
            "sass-loader"
          ],
          "allowedVersions": "^10.1.1"
        },
        {
          "packageNames": [
            "@vuex-orm/core"
          ],
          "allowedVersions": "^0.32.5"
        },
        {
          "packageNames": [
            "cypress"
          ],
          "allowedVersions": "6.4.0"
        },
        {
          "packageNames": [
            "postcss-loader"
          ],
          "allowedVersions": "^4.2.0"
        },
        {
          "matchPackagePatterns": [
            "^@myposter"
          ],
          "stabilityDays": 1
        },
        {
          "updateTypes": [
            "minor",
            "patch",
            "pin",
            "digest"
          ],
          "automerge": true
        },
        {
          "matchDepTypes": [
            "devDependencies",
            "peerDependencies"
          ],
          "extends": [
            ":semanticCommitType(chore)"
          ]
        },
        {
          "matchDepTypes": [
            "dependencies",
            "require"
          ],
          "extends": [
            ":semanticCommitType(update)"
          ]
        }
      ]
    }
  }
```

First thing not working is: dependencies starting with @myposter are also checked for stabilityDays: 5.
Second thing not working is: "dependencies" and "require" also get commit type: chore.

I have no idea why - everything else works as expected.

Thanks a lot for any answer!

Cheers, Tom

**Comment by viceice** [comment]
How do you publish and use your shared preset?

Npm hosted presets are deprecated.

Checkout our docs https://docs.renovatebot.com/config-presets/

**Comment by tom-agotchi-318** [comment]
Cannot find any information about npm hosted presets being deprecated?

https://docs.renovatebot.com/config-presets/#how-to-publish-preset-configs

This example is even with using npm 🤔 

But anyway, maybe we should change, but why are only two those rule not working?

**Reply by rarkins** [reply]
First of all, please make sure you're using the latest version of Renovate plus you are using the latest syntax for packageRules. e.g. `matchPackageNames` instead of `packageNames`. It hopefully isn't the cause but best to eliminate it as a possibility first. Then copy/paste your entire config again.

**Reply by tom-agotchi-318** [reply]
Shame on me.

I thought we had to be on latest renovate version, because it keeps itself up to date - but with "stabilityDays": 5 we never got updates because it is published very often.
After updating to latest version everything works as expected.

One following question: Any best practices if I want to keep 5 stabilityDays but receive an update at least once a month? That would solve the problem for packages that are release very often....

**Reply by rarkins** [reply]
If your main concern is upgrade frequency instead of upgrade stability, then putting Renovate on a monthly schedule would work best. That way you get a PR once per month and it will continue to get updated/modified in place for each release, but once you merge it then you won't get another PR until the next month.

e.g. package rule, matchPackageNames=["renovate"], schedule=["before 10am on the first day of the month"]

**Comment by tom-agotchi-318** [comment]
Well - my main concern is upgrade stability - but e.g. in case of renovate defining stabilityDays: 5 led to not updating for months because there are releases very frequently.

And since I don't want to scroll through the logs to determine which dependencies are out to date because of stabilityDays it would be great to configure something like: update daily if stabilityDays are met, but update at least once a month (while ignoring stabilityDays, to force being up to date).
Any possibilities achieving that?

**Reply by rarkins** [reply]
No. For a package like renovate it wouldn't make any stability improvement anyway because it would just end up picking a random release once a month too.

**Reply by tom-agotchi-318** [reply]
And any configuration possibility for me configure something like:
Once a month all packages should be updated to latest version even if they do not pass stabilityDays?

Or maybe you do have a different solution for avoiding frequently dependencies not being updated for a long time because of stabilityDays without creating a packageRules for each of those?

**Reply by rarkins** [reply]
No, only what I mentioned above is currently possible.

**Reply by tom-agotchi-318** [reply]
alright, then thanks a lot for your fast responses!

**Reply by rarkins** [reply]
There might be something new soon though. I'm working on a feature to perform "eager" filtering of releases based on stabilityDays so that you get a PR as soon as *any* of the newer releases pass the threshold. Anything not passing the threshold then gets suppressed until it does.

**Reply by tom-agotchi-318** [reply]
That would be pretty nice!
